supervized tests flaky

fairly easy to repro, `go test github.com/orbs-network/orbs-network-go/synchronization/supervized -count 100 -v` will make it die with the below output

```
=== RUN   TestShortLived_ReportsOnPanic
--- PASS: TestShortLived_ReportsOnPanic (0.00s)
=== RUN   TestLongLived_ReportsOnPanicAndRestarts
--- PASS: TestLongLived_ReportsOnPanicAndRestarts (0.00s)
=== RUN   TestShortLived_ReportsOnPanic
panic: foo

goroutine 8 [running]:
github.com/orbs-network/orbs-network-go/synchronization/supervized.TestLongLived_ReportsOnPanicAndRestarts.func1.1()
	/Users/levison/work/go/src/github.com/orbs-network/orbs-network-go/synchronization/supervized/supervisor_test.go:64 +0x55
github.com/orbs-network/orbs-network-go/synchronization/supervized.LongLived.func1(0xc4200ee460, 0x1312e40, 0xc420062840)
	/Users/levison/work/go/src/github.com/orbs-network/orbs-network-go/synchronization/supervized/supervisor.go:30 +0x27
created by github.com/orbs-network/orbs-network-go/synchronization/supervized.LongLived
	/Users/levison/work/go/src/github.com/orbs-network/orbs-network-go/synchronization/supervized/supervisor.go:28 +0x87
FAIL	github.com/orbs-network/orbs-network-go/synchronization/supervized	0.017s
```

--- a/synchronization/supervized/supervisor.go
+++ b/synchronization/supervized/supervisor.go
@@ -24,8 +24,8 @@ func ShortLived(logger Errorer, f func()) {
 
 // Runs f() in a goroutine; if it panics, logs the error and stack trace to the specified Errorer; if the provided Context isn't closed, re-runs f()
 func LongLived(ctx context.Context, logger Errorer, f func()) {
-	defer recoverPanics(logger)
 	go func() {
+		defer recoverPanics(logger)
 		for {
 			f()
 			if ctx.Err() != nil { // this returns non-nil when context has been closed via cancellation or timeout or whatever

--- a/synchronization/supervized/supervisor_test.go
+++ b/synchronization/supervized/supervisor_test.go
@@ -58,7 +58,7 @@ func TestLongLived_ReportsOnPanicAndRestarts(t *testing.T) {
 	require.NotPanicsf(t, func() {
 		LongLived(ctx, logger, func() {
 			count++
-			if count > 3 {
+			if count > 10 {
 				cancel()
 			}
 			panic("foo")

